tests/Microsoft.Identity.Test.Unit/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -594,6 +594,50 @@ public void WithClientCapabilitiesViaOptions()
             CollectionAssert.AreEquivalent(new string[] { "cp1", "cp2" }, app.AppConfig.ClientCapabilities.ToList());
         }
 
+        [TestMethod]
+        // bug https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2543
+        public void AuthorityWithTenant()
+        {
+            var options = new PublicClientApplicationOptions();
+            options.ClientId = TestConstants.ClientId;            
+
+            var app1 = PublicClientApplicationBuilder
+                .CreateWithApplicationOptions(options)
+                .WithTenantId(TestConstants.TenantId)                            
+                .WithAuthority("https://login.microsoftonline.com/common")       
+               .Build();
+
+            Assert.AreEqual($"https://login.microsoftonline.com/{TestConstants.TenantId}/", app1.Authority);
```

Comment:
nit: `$"https://login.microsoftonline.com/{TestConstants.TenantId}/"` can be changed to `TestConstants.AuthorityTenant`